src/week2/oop/Parser.java

```diff
@@ -0,0 +1,40 @@
+package week2.oop;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Created by bogdan on 3/13/2017.
+ */
+public class Parser {
+
+    private static final String QUADRATIC_EQUATION = "^([+-]?\\d+|[+-]?\\d+\\.\\d+)x\\^2\\s([+-])\\s(\\d+|\\d+\\.\\d+)x"
+                                                        +"\\s([+-])\\s(\\d+|\\d+\\.\\d+)\\s=\\s0$";
+    private static final String LINEAR_EQUATION = "^([+-]?\\d+|[+-]?\\d+\\.\\d+)x\\s([+-])\\s(\\d+|\\d+\\.\\d+)\\s=\\s0$";
+
+    public int parse(String s){
```

Comment:
got it. 
It's more better to create enum with parsing results. 
Like 
public EquationType parse(string s)

where 
enum EquationType{LINEAR, QUADRATIC}
and return null if parsing is impossible